core-ui/src/state/navigation/filters/filterPermittedNodes.ts

```diff
@@ -0,0 +1,24 @@
+import { doesUserHavePermission } from './permissions';
+import { NavNode } from '../../types';
+import { PermissionSet } from '../../permissionSetsAtom';
+
+export const filterPermittedNodes = (
+  nodes: NavNode[],
+  permissionSet: PermissionSet[],
+) => {
+  return nodes.filter(node => isResourcePermitted(node, permissionSet));
+};
+
+const isResourcePermitted = (node: NavNode, permissionSet: PermissionSet[]) => {
+  const resourceGroupAndVersion = `${node.apiGroup}${node.apiGroup ? '/' : ''}${
+    node.apiVersion
```

Comment:
is this double `node.apiGroup` necessary?

Reply:
we want to add / if apiGroup is present. Currently can't think of other solution. any ideas?